oresat_dxwifi/camera/camera.py

```diff
@@ -119,116 +63,144 @@ def __init__(
+            logger.info("Created new directory: {}".format(path))
 
-    def delete_frames(self):
-        # If requested, delete frames
+    def _delete_frames(self) -> None:
+        """If frame delete variable is true, deletes frames in image output directory"""
         if self.are_frames_deleted:
-            shutil.rmtree(self.image_output_directory_path)
-            print("Deleted raw frames at: {}".format(self.image_output_directory_path))
-
-    def create_videos(self):
-        # Figure out how many times to loop
-        num_loops = cam_interface.calculate_number_of_loops()
-
-        # Use a list to keep track of asynchronous ffmpeg calls
-        procs = []
-
-        # Make videos
-        for i in range(num_loops):
-            # Create strings
-            img_dir = os.path.join(self.image_output_directory_path, "{:04d}".format(i))
-            vid_name = os.path.join(self.video_output_directory_path, "output{:04d}.mp4".format(i))
-
-            # Make directory for frames
-            os.mkdir(img_dir)
-
-            # Make commands
-            capture_command = "{} {} {} {} {} {} {}".format(
-                self.camera_binary_path,
-                self.device_path,
-                self.horizontal_video_resolution,
-                self.vertical_video_resolution,
-                self.frames_per_second,
-                self.seconds_per_video,
-                img_dir,
+            shutil.rmtree(self.image_output_directory)
+            logger.info("Deleted raw frames at: {}".format(self.image_output_directory))
+
+    def create_videos(self) -> None:
+        """Captures and encodes videos
+
+        Raises:
+            CameraInterfaceError: "Seconds per video does not evenly divide video duration".
+            CameraInterfaceError (directory_error): Something went wrong with clearing the directories
+            CameraInterfaceError (capture_error): Something went wrong with capture or encoding
+        """
+        if not self._check_seconds_per_video():
+            raise CameraInterfaceError(
+                "Seconds per video does not evenly divide video duration"
             )
-            ffmpeg_command = "ffmpeg -framerate {} -i {} {} {}".format(
-                args.fps, os.path.join(img_dir, "frame%04d.ppm"), self.FFMPEG_CONSTANTS, vid_name
-            )
-
-            # Call capture (blocking)
-            print("Capturing frames for video {} of {}.".format(i + 1, num_loops))
-            subprocess.call(capture_command.split())
 
-            # Call ffmpeg (non-blocking)
-            print("Starting encoding of video {} of {}.".format(i + 1, num_loops))
-            procs.append(subprocess.Popen(ffmpeg_command.split()))
+        try:
+            self._fresh_dir(self.image_output_directory)
+            self._fresh_dir(self.video_output_directory)
+        except Exception as directory_error:
+            raise CameraInterfaceError(directory_error)
```

Comment:
Valid. At the time I just didn't want to rework fresh_dir and get straight to the camera system working. The only way it fails in theory I think is the directories don't exist.  I think it's low priority for now since the message goes through.

Reply:
Just add TODO comments